### DISCUSSION TITLE
How can we make this game more challenging and funner?

### DISCUSSION THREAD

**Discussion by rajatnai49** [rajatnai49/sneakySnake#3] [Q&A] [answered]
You can give your suggestion about how can we make this game more challenging and fun to play.

**Comment by kunjgit** [comment, marked as answer]
To make this game effiecient what you can doo is hash down the snake part so it will be directly available without traversing the whole DS.
And then in app.js
you can make improvisation by making the game more challanging to play for an example you can have the difficulty modes and based upon that you can rearrange the position of snake .
You can add practise session where user can themselves fix the positions of the elements and try it themselves 


One optimization you can provide random function extra info about the range in which conflict will ocurre the least

```
function checkCollision() {
    let cox=-1
    let coy=-1

     for (let i = 0; i < snakeParts.length; i++) {
        let part = snakeParts[i]
        cox=part.x
        coy=part.y
        if(cox==appleX && coy==appleY){
        appleX = Math.floor(Math.random() * tileCount);
        appleY = Math.floor(Math.random() * tileCount);
        ctx.checkCollison()

        }
    }

    if (appleX == headX && appleY == headY) {
        appleX = Math.floor(Math.random() * tileCount);
        appleY = Math.floor(Math.random() * tileCount);
```
